src/js/stores/font.js

```diff
@@ -138,27 +138,41 @@ define(function (require, exports, module) {
                 psName = textStyle.postScriptName,
                 fontObj = this._postScriptMap.get(psName, null);
                 
-            if (!fontObj) {
-                // FIXME: What about missing fonts?
-                throw new Error("The font for layer is not available!");
-            }
+            // If it's mixed, or there is no fontObj, we skip adding these values to the type object
+            // adbeFont, fontFamily, fontStyle, fontWeight
+            if (fontObj) {
+                var family = fontObj.family,
+                    fontRecord = this._familyMap.get(family),
+                    psObj = fontRecord.get(fontObj.font, null);
+
+                if (!psObj) {
+                    throw new Error("The font for layer is not available!");
```

Comment:
I don't want this function to return null, I want it to return whatever is available in that layer's characterstyle in an object. For mixed state layers, this will only be the common properties. I'll at least log a warning there and skip that property.